### PR TITLE
feat(awslambda): add new check `awslambda_function_vpc_multi_az`

### PR DESCRIPTION
**Pull request by puchy22**
### Context

New check that ensures AWS Lambda functions connected to a Virtual Private Cloud (VPC) are distributed across multiple Availability Zones (AZs). This is crucial for maintaining high availability and resilience, as deploying Lambda functions in a single AZ can lead to a single point of failure, potentially disrupting operations if that AZ becomes unavailable.

### Description

- [x]  Modify service to add Subnets to AWS Lambda Functions
- [x]  Test service changes
- [x]  Add configurable parameter for the check
- [x]  Add check logic
- [x]  Add unit testing for the check

### Checklist

- Are there new checks included in this PR? Yes
    - If so, do we need to update permissions for the provider? No.
- [x] Review if the code is being covered by tests.
- [x] Review if code is being documented following this specification https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json [hunk 1]
```diff
@@ -0,0 +1,30 @@
+{
+  "Provider": "aws",
+  "CheckID": "awslambda_function_vpc_is_in_multiple_az",
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json]:
```suggestion
  "CheckID": "awslambda_function_vpc_multi_az",
```

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+class awslambda_function_vpc_is_in_multiple_az(Check):
+    def execute(self) -> List[Check_Report_AWS]:
+        findings = []
+        LAMBDA_MINIMAL_AZ = awslambda_client.audit_config.get("lambda_minimal_az", 2)
+        for function_arn, function in awslambda_client.functions.items():
+            # only proceed if check "awslambda_function_inside_vpc" did not run or did not FAIL to avoid to report that the function is not inside a VPC twice
+            if not awslambda_client.is_failed_check(
+                awslambda_function_inside_vpc.__name__,
+                function_arn,
+            ):
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: I cannot reproduce this logic, please take it a look and add a test.

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+                report.status = "FAIL"
+                report.status_extended = (
+                    f"Lambda function {function.name} is not inside a VPC."
+                )
+
+                if function.vpc_id:
+                    function_availability_zones = set(
+                        [
+                            vpc_client.vpc_subnets[subnet_id].availability_zone
+                            for subnet_id in function.subnet_ids  # BUG: if prowler executes this check with -f option, it will not have the subnets in the vpc_client.vpc_subnets dict
```
Comment (MrCloudSec) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: Why this? With -f, the VPCs of that region should have their subnets.

Comment (MrCloudSec) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: I could reproduce this use case and it worked.

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json (hunk 1 above)
Comment (puchy22) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json]: The rest of the new checks related to multiple Availability Zones are called `*_multiple_az`, don't you think they should still be called that to keep the format?

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py (hunk 3 above)
Comment (puchy22) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: You are right the comment is wrong, it is a bug with the parameter `--resource-arn`. When the parameter is pointing to a lambda function the every attribute of VPC service seems to be empty.

I think that the reason is the following lines:
```
                    if not self.audit_resources or (
                        is_resource_filtered(subnet["SubnetArn"], self.audit_resources)
                    ):
```
Command used:
`prowler aws --log-level ERROR --verbose -f eu-west-1 -c awslambda_function_vpc
_is_in_multiple_az`

Comment (puchy22) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: I changed the code to manage the exception.

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py (hunk 2 above)
Comment (puchy22) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: Added the logic of failed check to other check and added one test to ensure everything is working as expected.

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json (hunk 1 above)
Comment (jfagoagas) [reply prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json]: I think `multi_az` is more aligned with the cloud models.

File: docs/tutorials/configuration_file.md [hunk 4]
```diff
@@ -48,6 +48,7 @@ The following list includes all the AWS checks with configurable variables that
 | `eks_control_plane_logging_all_types_enabled`                 | `eks_required_log_types`                         | List of Strings |
 | `eks_cluster_uses_a_supported_version`                        | `eks_cluster_oldest_version_supported`           | String          |
 | `elbv2_is_in_multiple_az`                                     | `elbv2_min_azs`                                  | Integer         |
+| `awslambda_function_vpc_is_in_multiple_az`                    | `lambda_min_azs`                                 | Integer         |
```
Comment (jfagoagas) [inline docs/tutorials/configuration_file.md]: Is this correct?

File: prowler/config/config.yaml [hunk 5]
```diff
@@ -101,6 +101,8 @@ aws:
       "ruby2.5",
       "ruby2.7",
     ]
+  # aws.awslambda_function_vpc_is_in_multiple_az
```
Comment (jfagoagas) [inline prowler/config/config.yaml]: Review this name too.

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  "CheckTitle": "Check if AWS Lambda Function VPC is Distributed Across Multiple Availability Zones",
```
Comment (jfagoagas) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.metadata.json]: Distributed or Deployed?

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+from prowler.providers.aws.services.awslambda.awslambda_function_inside_vpc.awslambda_function_inside_vpc import (
+    awslambda_function_inside_vpc,
+)
+from prowler.providers.aws.services.vpc.vpc_client import vpc_client
+
+
+class awslambda_function_vpc_is_in_multiple_az(Check):
+    def execute(self) -> List[Check_Report_AWS]:
+        findings = []
+        LAMBDA_MINIMAL_AZ = awslambda_client.audit_config.get("lambda_minimal_az", 2)
```
Comment (jfagoagas) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: `lambda_minimal_az` or `lambda_min_azs`? 😅

File: prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+                )
+
+                if function.vpc_id:
+                    function_availability_zones = set(
+                        [
+                            getattr(
+                                vpc_client.vpc_subnets.get(subnet_id, None),
+                                "availability_zone",
+                                None,
+                            )  # BUG: if prowler executes this check with --resource-arn option, it will not have the subnets in the vpc_client.vpc_subnets dict (Exception managed in the getattr function)
```
Comment (jfagoagas) [inline prowler/providers/aws/services/awslambda/awslambda_function_vpc_is_in_multiple_az/awslambda_function_vpc_is_in_multiple_az.py]: Is this a `bug`? Could you add a test to check this behaviour?

**Review by jfagoagas — DISMISSED** [review]

**Review by MrCloudSec — APPROVED** [review]

**Pull request merged**